.github/workflows/path_detection.yml

```diff
@@ -0,0 +1,31 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+name: Check for missing Dockerfile paths in repo comps
+
+on:
+  pull_request:
+    branches: [main]
+    types: [opened, reopened, ready_for_review, synchronize]
+
+jobs:
+  check-dockerfile-paths:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repo GenAIExamples
+      uses: actions/checkout@v4
+
+    - name: Clone repo GenAIComps
+      run: |
+        git clone https://github.com/opea-project/GenAIComps.git
+
+    - name: Check for missing Dockerfile paths in GenAIComps
+      run: |
+        grep -Ern 'docker build .* -f comps/.+/Dockerfile' --include='*.md' . | while IFS=: read -r file line content; do
+            dockerfile_path=$(echo "$content" | awk -F '-f ' '{print $2}' | awk '{print $1}')
+            if [[ ! -f "../GenAIComps/${dockerfile_path}" ]]; then
```

Comment:
I wonder the folder `GenAICompos` cloned in previous step is a child of the folder of `GenAIExamples` or a sibling of folder `GenAiExamples`. If `GenAICompos` is the child of `GenAIExamples`, `../GenAIComps` is wrong. Please confirm. Thank you.